xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/suggest/xwiki.selectize.js

```diff
@@ -135,6 +135,12 @@ define('xwiki-selectize', ['jquery', 'selectize', 'xwiki-events-bridge'], functi
         var wrapper = this.$wrapper;
         wrapper.addClass(this.settings.loadingClass);
       }
+    },
+    onItemAdd: function(value, item) {
+      item.find('a.xwiki-selectize-option-label').click(function(event) {
+        // Clicking on the label should select the option not follow the link.
+        event.preventDefault();
+      });
```

Comment:
You don't need to add the event listener on each suggestion, each time we add a suggestion. You can add the event listener **once** on a parent element that is always there and use the 'selector' to filter where the event comes from (since the event is propagated upwards), see https://api.jquery.com/on/#on-events-selector-data-handler .

Reply:
@mflorea I did as you suggested which is a way better way of doing what we need and solves the issue.